(Docs+) Flush out Resource+Task troubleshooting (#111773)
elastic/elasticsearch: docs/reference/troubleshooting/common-issues/high-cpu-usage.asciidoc

related to the thread pool. For example, if the `search` thread pool is
depleted, {es} will reject search requests until more threads are available.

> CPU degradation frequently occurs related to a <<data-tiers,data tier>>'s traffic,
> potentially being <<hotspotting,hot spotted>>.

Reviewer: this sentence is confusing to me. couple of things:

- the cpu isn't degraded, is it? it's just experiencing high usage? 
- data tier stuff needs a little clarification. perhaps:

```suggestion
You might experience high CPU usage if a <<data-tiers,data tier>>, and therefore the nodes assigned to that tier, is experiencing more traffic than other tiers. This imbalance in resource utilization is also known as <<hotspotting,hot spotting>>.
```

[discrete]
[[diagnose-high-cpu-usage]]
==== Diagnose high CPU usage

**Check CPU usage**

> Current CPU usage per node can be polled from the <<cat-nodes,cat nodes API>>:

Reviewer: ```suggestion
You can check the CPU usage per node using the <<cat-nodes,cat nodes API>>:
```

// tag::cpu-usage-cat-nodes[]
[source,console]
----
GET _cat/nodes?v=true&s=cpu:desc
----

The response's `cpu` column contains the current CPU usage as a percentage.

> The `name` column contains the node's name. Elevated but transient `cpu` is
> normal, but if `cpu` is elevated for an extended duration it should be
> investigated.

Reviewer: ```suggestion
The `name` column contains the node's name. Elevated but transient CPU usage is
normal. However, if CPU usage is elevated for an extended duration, it should be
investigated.
```

[source,console]
----
GET _nodes/hot_threads
----
// TEST[s/\/my-node,my-other-node//]

> This API returns a breakdown of any hot threads in plain text. High CPU usage
> frequently correlates to <<task-queue-backlog,particular tasks and/or their
> backlog>>.

Reviewer: [we can just use or](https://www.collinsdictionary.com/dictionary/english/inclusive-or) :)

```suggestion
This API returns a breakdown of any hot threads in plain text. High CPU usage
frequently correlates to <<task-queue-backlog,a long-running task, or a
backlog of tasks>>.
```
